pkg/quarterdeck/auth.go

```diff
@@ -448,3 +449,96 @@ func (s *Server) Refresh(c *gin.Context) {
 	}))
 	c.JSON(http.StatusOK, out)
 }
+
+// VerifyEmail verifies a user's email address by validating the token in the request.
+// This endpoint is intended to be called by frontend applications after the user has
+// followed the link in the verification email. If the token is already verified this
+// endpoint returns a 202 Accepted response.
+func (s *Server) VerifyEmail(c *gin.Context) {
+	var (
+		req *api.VerifyRequest
+		err error
+	)
+
+	// Get the token from the POST request
+	if err = c.BindJSON(&req); err != nil {
+		log.Warn().Err(err).Msg("could not parse verify email request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse request"))
+		return
+	}
+
+	if req.Token == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("missing token in request"))
+		return
+	}
+
+	// Look up the user by the token
+	var user *models.User
+	if user, err = models.GetUserByToken(c, req.Token); err != nil {
+		if errors.Is(err, models.ErrNotFound) {
+			c.JSON(http.StatusUnauthorized, api.ErrorResponse("invalid token"))
```

Comment:
```suggestion
			c.JSON(http.StatusBadRequest, api.ErrorResponse("invalid token"))
```

I think a 400 is probably slightly better here than a 401 for security purposes.